idf-rust.Containerfile

```diff
@@ -4,35 +4,64 @@ FROM debian:${VARIANT}
-ARG INSTALL_RUST_TOOLCHAIN=install-rust-toolchain.sh
+ARG ESP_BOARD=all
+ARG GITHUB_TOKEN
+ARG XTENSA_VERSION=latest
+
 # Install dependencies
 RUN apt-get update \
-    && apt-get install -y git curl gcc clang ninja-build libudev-dev unzip xz-utils \
-    python3 python3-pip python3-venv libusb-1.0-0 libssl-dev pkg-config libtinfo5  libpython2.7 \
+    && apt-get install -y git curl gcc clang ninja-build unzip libudev-dev tar xz-utils \
+    python3 python3-pip python3-venv libusb-1.0-0 libssl-dev pkg-config libpython2.7 \
     && apt-get clean -y && rm -rf /var/lib/apt/lists/* /tmp/library-scripts
-# Set user
+
+# Set users
 RUN adduser --disabled-password --gecos "" ${CONTAINER_USER}
 USER ${CONTAINER_USER}
 WORKDIR /home/${CONTAINER_USER}
-# Install rust toolchain(s), extra crates and esp-idf.
+
+# Install rustup
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- \
+    --default-toolchain none -y --profile minimal
+
+# Update envs
 ENV PATH=${PATH}:/home/${CONTAINER_USER}/.cargo/bin
-ADD --chown=${CONTAINER_USER}:${CONTAINER_GROUP} \
-    https://github.com/esp-rs/rust-build/releases/download/v${XTENSA_TOOLCHAIN_VERSION}/${INSTALL_RUST_TOOLCHAIN} \
-    ${INSTALL_RUST_TOOLCHAIN}
-RUN chmod a+x ${INSTALL_RUST_TOOLCHAIN} \
-    && ./${INSTALL_RUST_TOOLCHAIN} \
-    --extra-crates "ldproxy cargo-espflash cargo-generate sccache" \
-    --build-target "${ESP_BOARD}" \
-    --nightly-version "${NIGHTLY_TOOLCHAIN_VERSION}" \
-    --esp-idf-version "${ESP_IDF_VERSION}" \
-    --minified-esp-idf "YES" \
-    --export-file  ${HOME}/export-esp.sh
-# Activate ESP-IDF and Xtensa Rust toolchain environment
-RUN echo "source ${HOME}/export-esp.sh" >> ~/.bashrc
+
+# Install extra crates
+RUN ARCH=$($HOME/.cargo/bin/rustup show | grep "Default host" | sed -e 's/.* //') && \
+    curl -L "https://github.com/esp-rs/espup/releases/latest/download/espup-${ARCH}" -o "${HOME}/.cargo/bin/espup" && \
+    chmod u+x "${HOME}/.cargo/bin/espup" && \
+    curl -L "https://github.com/esp-rs/embuild/releases/latest/download/ldproxy-${ARCH}.zip" -o "${HOME}/.cargo/bin/ldproxy.zip" && \
+    unzip "${HOME}/.cargo/bin/ldproxy.zip" -d "${HOME}/.cargo/bin/" && \
+    rm "${HOME}/.cargo/bin/ldproxy.zip" && \
+    chmod u+x "${HOME}/.cargo/bin/ldproxy" && \
+    curl -L "https://github.com/bjoernQ/esp-web-flash-server/releases/latest/download/web-flash-${ARCH}.zip" -o "${HOME}/.cargo/bin/web-flash.zip" && \
+    unzip "${HOME}/.cargo/bin/web-flash.zip" -d "${HOME}/.cargo/bin/" && \
+    rm "${HOME}/.cargo/bin/web-flash.zip" && \
+    chmod u+x "${HOME}/.cargo/bin/web-flash"
+
+# Install Rust toolchain for our ESP_BOARD
+RUN if [ -n "${GITHUB_TOKEN}" ]; then export GITHUB_TOKEN=${GITHUB_TOKEN}; fi && \
+    version="" && \
+    if [ "${XTENSA_VERSION}" != "latest" ];then version="--toolchain-version ${XTENSA_VERSION}"; fi && \
+    ${HOME}/.cargo/bin/espup install\
+    --targets "${ESP_BOARD}" \
+    --log-level debug \
+    --profile-minimal \
```

Comment:
```suggestion
```

Yesterday, I release `[espup v0.3.0`](https://github.com/esp-rs/espup/releases/tag/v0.3.0) which deprecates the `--profile-minimal` flag as it's now minimal by default, hence we will need to remove it.